qiita_db/meta_util.py

```diff
@@ -118,6 +116,40 @@ def get_accessible_filepath_ids(user):
                              in SampleTemplate(study_id).get_filepaths()]
             filepath_ids.update(sample_fp_ids)
 
+    # Next, the public processed data
+    processed_data_ids = ProcessedData.get_by_status('public')
+    for pd_id in processed_data_ids:
```

Comment:
This seems amazingly expensive, especially if this is going to be used to check file access and needs to happen every download. Can you create a PrepTemplate(x).associated_files or some other way of getting these out with far less database calls.